src/OpenTelemetry/Logs/OpenTelemetryLogger.cs

```diff
@@ -26,15 +25,12 @@ namespace OpenTelemetry.Logs
     internal class OpenTelemetryLogger : ILogger
     {
         private readonly string categoryName;
+        private readonly OpenTelemetryLoggerProvider provider;
 
-        internal OpenTelemetryLogger(string categoryName, OpenTelemetryLoggerOptions options)
+        internal OpenTelemetryLogger(string categoryName, OpenTelemetryLoggerProvider provider)
```

Comment:
it feels a bit awkward that the Logger has a reference to the Provider itself...Can we instead keep the processor as part of the a different class or `OpenTelemetryLoggerOptions` itself?

Reply:
I think we have to back-reference since the LoggerProvider owns the processor chain, and it is subject to change after ctor.

Reply:
Do we have to make LoggerProvider own the processor chain? Can it instead be owned by the `OpenTelemetryLoggerOptions ` ?

My mental model:
There is OTelProvider. And there is `OpenTelemetryLoggerOptions`.
If i want to configure anything on this provider, I use `OpenTelemetryLoggerOptions`.
If I want to add a new processor for this OTelProvider, I use `OpenTelemetryLoggerOptions`.

(checking to see if OpenTelemetryLogger can hold a reference to the options, which holds the processor)

Reply:
> Do we have to make LoggerProvider own the processor chain?

We don't have to.

> Can it instead be owned by the `OpenTelemetryLoggerOptions ` ?

I thought about it as well and decided it was not a good way to go. Couple reasons:

1. The following snippet is more aligned with the OpenTelemetry tracing experience:
    ```csharp
    using var loggerProvider = Sdk.CreateLoggerProviderBuilder()
        .AddProcessor(new MyProcessor())
        .Build();
    var logger = loggerProvider.CreateLogger("MyLogger");
    ```

2. `Options` is not an OpenTelemetry concept, the only reason I added support for it is to enable .NET dependency injection and factory paradigm. I don't have a strong opinion though.